skyplane/api/dataplane.py

```diff
@@ -39,14 +39,17 @@ class Dataplane:
 
     def __init__(
         self,
+        clientid: str,
         topology: ReplicationTopology,
         provisioner: "Provisioner",
         transfer_config: TransferConfig,
     ):
+        self.clientid = clientid
         self.topology = topology
         self.src_region_tag = self.topology.source_region()
         self.dst_region_tag = self.topology.sink_region()
-        self.max_instances = int(len(self.topology.gateway_nodes) / 2)
+        regions = Counter([node.region.split(":")[1] for node in self.topology.gateway_nodes])
```

Comment:
In theory there shouldn't be collisions between clouds but maybe should we index it as ("aws", "us-east-1") instead?